src/lib/paginator/paginator.ts

```diff
@@ -101,12 +115,14 @@ export class MatPaginator extends _MatPaginatorBase implements OnInit, OnDestroy
 
   /** The set of provided page size options to display to the user. */
   @Input()
-  get pageSizeOptions(): number[] { return this._pageSizeOptions; }
-  set pageSizeOptions(value: number[]) {
-    this._pageSizeOptions = (value || []).map(p => coerceNumberProperty(p));
+  get pageSizeOptions(): (number | PageSizeOption)[] { return this._pageSizeOptions; }
```

Comment:
Would it be easier to add a `@Input() pageSizeOptionText` function which of type `(number) => string` and the default value is `(o) => '${o}'`? 

And in html we can directly use `{{pageSizeOptionText(pageSizeOption)}}`, without converting every number to a `PageSizeOption` object.